src/magic_folder/web.py

```diff
@@ -98,17 +98,45 @@ def getChildWithDefault(self, path, request):
 
 
 @attr.s
-class V1MagicFolderAPI(Resource, object):
+class APIv1(Resource, object):
     """
-    The root of the ``/v1`` HTTP API hierarchy.
+    Implement the ``/v1`` HTTP API hierarchy.
 
-    :ivar (unicode -> MagicFolder) _get_magic_folder: A function that looks up
-        a magic folder by its nickname.
+    :ivar MagicFolderServiceState _magic_folder_state: The Magic Folder state
+        to serve.
     """
-    _get_magic_folder = attr.ib()
+    _magic_folder_state = attr.ib()
 
     def __attrs_post_init__(self):
         Resource.__init__(self)
+        self.putChild(b"magic-folder", MagicFolderAPIv1(self._magic_folder_state))
+
+
+@attr.s
+class MagicFolderAPIv1(Resource, object):
+    """
+    Implement the ``/v1/magic-folder`` HTTP API hierarchy.
+
+    :ivar MagicFolderServiceState _magic_folder_state: The Magic Folder state
+        to serve.
+    """
+    _magic_folder_state = attr.ib()
+
+    def __attrs_post_init__(self):
+        Resource.__init__(self)
```

Comment:
Should this be a `super()` incantation instead since we have `object` too?

Reply:
Hm.  I don't know what would happen with `super()` here - specifically because it's not `MagicFolderAPIv1.__init__` that's upcalling `Resource.__init__`.  Maybe it's fine.  I also just realized I was afraid of calling the attrs-generated `__init__` twice but on further thought that seems impossible because it's generated on *this* class not on a super class...?  I'll try changing it and see if something blows up.

Reply:
It seems to work but I'm having trouble convincing myself it's really the right thing.  This is specifically trying to call `Resource.__init__` because the attr.s-generated `__init__` didn't call it.  `Resource.__init__` is pretty clearly calling the right thing.  `super(...).__init__` is a lot less clear to me.  This is basically just a different way to say what I said in my previous comment, I guess.

It seems like the ideal would be for the attr.s-generated `__init__` to make this super() call itself.  It wouldn't surprise me if there are boogeymen hiding behind that idea though...

I'm going to leave this for now but happy to revisit it later.